### PR TITLE
Refactor ContractCallSystemPrecompileTest

### LINKED ISSUE #8453: Refactor `ContractCallSystemPrecompileTest`

### PR DESCRIPTION
**Pull request by kselveliev**
**Description**:
This PR refactors ContractCallSystemPrecompileTest to use the web3j plugin.
Historical test are moved into a new class ContractCallSystemPrecompileHistoricalTest.

This PR modifies;
Refactored ContractCallSystemPrecompileTest to use the web3j approach
Moved historical tests into ContractCallSystemPrecompileHistoricalTest.
Added a new method in AbstractContractCallServiceTest that asserts that estimate gas fails with expected 
exception and error message.

Fixes #8453 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/OpcodeServiceTest.java [hunk 1]
```diff
@@ -322,7 +1167,7 @@ void callWithContractResultNotFoundExceptionTest() {
             contractEntityId = systemExchangeRateContractPersist();
 
             final TransactionIdOrHashParameter transactionIdOrHash = setUp(
-                    ContractCallSystemPrecompileTest.ExchangeRateFunctions.TINYBARS_TO_TINYCENTS,
+                    ContractCallSystemPrecompileHistoricalTest.ExchangeRateFunctions.TINYBARS_TO_TINYCENTS,
                     TransactionType.CONTRACTCALL,
```
Comment (kselveliev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/OpcodeServiceTest.java:1171]: This is the only change added to this class.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/OpcodeServiceTest.java [hunk 2]
```diff
@@ -340,7 +1185,7 @@ void callWithTransactionNotFoundExceptionTest() {
             contractEntityId = systemExchangeRateContractPersist();
 
             final TransactionIdOrHashParameter transactionIdOrHash = setUp(
-                    ContractCallSystemPrecompileTest.ExchangeRateFunctions.TINYCENTS_TO_TINYBARS,
+                    ContractCallSystemPrecompileHistoricalTest.ExchangeRateFunctions.TINYCENTS_TO_TINYBARS,
                     TransactionType.CONTRACTCALL,
```
Comment (kselveliev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/OpcodeServiceTest.java:1189]: This is the only change added to this class.

**Comment by kselveliev** [comment]
Only 2 lines were updated in OpcodeServiceTest. The rest of the changes are formattings applied by the spotlessApply script.

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallSystemPrecompileHistoricalTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package com.hedera.mirror.web3.service;
```
Comment (edwin-greene) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallSystemPrecompileHistoricalTest.java:1]: Needs a copyright license header.

Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallSystemPrecompileHistoricalTest.java:17]: Comment addressed

**Review by steven-sheehy — APPROVED** [review]

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**